net-misc/httpie/httpie-1.0.2.ebuild

```diff
@@ -0,0 +1,37 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python2_7 python3_{4,5,6,7} )
+
+inherit bash-completion-r1 distutils-r1
+
+DESCRIPTION="Modern command line HTTP client"
+HOMEPAGE="https://httpie.org/ https://pypi.org/project/httpie/"
+SRC_URI="https://github.com/jakubroztocil/httpie/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+RDEPEND="dev-python/pygments[${PYTHON_USEDEP}]
+	>=dev-python/requests-2.19.1[${PYTHON_USEDEP}]"
+DEPEND="dev-python/setuptools[${PYTHON_USEDEP}]
+	test? (
+		${RDEPEND}
+		dev-python/pytest-httpbin
```

Comment:
This should have `[${PYTHON_USEDEP}]` because test imports it

https://github.com/jakubroztocil/httpie/blob/b88e88d2e373e2113015cb96c5e6edce9d2f1a91/tests/test_ssl.py#L3-4

repoman/ci screams at you because you claim to support python3.7 but pytest-httpbin does not, so if a user enables python3.7 support AND test will not be able to satisfy that import, because there will be no pytest-httpbin module to import.
you probably did not run test with python3_7 (and all deps with it) enabled. you should generally run test will all implementations that ebuild lists as supported ( I know it can be a PITA =))

there are 2 ways to deal with it
* drop py3.7 support for now and wait for or help with ->
* bumps for `httpbin` and `pytest-httpbin` with proper support for `python3.7`. both are oudated and a bump is needed for sure, and quickly looking at git master it looks like 3.7 support was added recently to travis. sometimes bumping python stuff is never-ending chain of bumps, because all new stuff requires new stuff.

Reply:
I don't consider it reasonable to drop Python 3.7 support only in order to get tests to work, and neither is waiting for test support packages to get updated. The more people hold back on Python 3.7 the more difficult to have this version supported.

I tried to bump `httpbin` to 0.6.1 locally, but my ebuild won't run:

```
The following USE changes are necessary to proceed:
 (see "package.use" in the portage(5) man page for more details)
# required by dev-python/httpbin (argument)
>=dev-python/httpbin-0.6.1 -python_targets_python3_7
```

Given my `make.conf` setting of `PYTHON_TARGETS="python2_7 python3_6 python3_7"` and `PYTHON_COMPAT=( python2_7 python3_{4,5,6,7} pypy pypy3 )` in the ebuild, I don't know why it is not working.

I made the other changes you requested, so all that remains is that tests won't run with Python 3.7. Personally, I can live with that.

Reply:
well, I can't commit as is. It's broken even if you can live with it. gentoo tree will be broken.

please drop 3.7 for now and add `[${PYTHON_USEDEP}]`, it's trivial to add it back later after everything is satisfied.
many packages miss 3.7 support, we'll get there, just later.

Reply:
That hack it not acceptable.
and you are still missing `[${PYTHON_USEDEP}]` for http-bin

last time I checked updating all required test deps will involve bumping ~10 ebuilds, packaging 5 new ebuilds. I haven't looked at  testdeps, but total number can be close to 30 ebuilds in worst case.
I just joined `python` project in gentoo and probably can help with it.

I still suggest merging this PR without py3.7 support.
if you still want to use 3.7 you can use `PYTHON_COMPAT_OVERRIDE="python3_7"` for this package in your portage settings locally.
nothing wrong with dropping 3.7 support. 3.6 is going to stay for long time. httpie is not a package where it really technically matters to have 3.7 support.

Reply:
> I haven't looked at testdeps, but total number can be close to 30 ebuilds in worst case.

Ouch.

> I still suggest merging this PR without py3.7 support.

As much as it irks me, I have modified the pull request accordingly. I also added PYTHON_USEDEP and squashed the commits. Hopefully I did not forget anything this time.

Reply:
thanks! should me merged by bot soon.
don't worry, we'll get packages to 3.7 eventually. it just takes time for both upstream and downstream to iron out everything.

Reply:
> don't worry, we'll get packages to 3.7 eventually.

You are right to state that it does not matter much for HTTPie. It just bothers me that support for Python 3.7 seems so slow to me. I routinely use 3.7 features in my own code, but fortunately I don't require Gentoo packaging for that. Please let me know if you see a way for me to help with getting packages ready for Python 3.7.